lock-api/src/test/java/com/palantir/lock/client/CoalescingTransactionStarterTest.java

```diff
@@ -0,0 +1,201 @@
+        CoalescingTransactionStarter.consumer(lockLeaseService).accept(elements);
+        return Futures.getUnchecked(Futures.allAsList(Lists.transform(elements, BatchElement::result)));
+    }
+
+    private void assertThatStartTransactionResponsesAreUnique(
+            List<StartIdentifiedAtlasDbTransactionResponse> responses) {
+        assertThat(responses)
+                .as("Each response should have a different immutable ts lock token")
+                .extracting(response -> response.immutableTimestamp().getLock().getRequestId())
+                .doesNotHaveDuplicates();
+
+        assertThat(responses)
+                .as("Each response should have a different start timestamp")
+                .extracting(response -> response.startTimestampAndPartition().timestamp())
+                .doesNotHaveDuplicates();
+    }
+
+    private void assertDerivableFromBatchedResponse(
+            StartIdentifiedAtlasDbTransactionResponse startTransactionResponse,
+            StartTransactionResponseV4 batchedStartTransactionResponse) {
+
+        assertThat(startTransactionResponse.immutableTimestamp().getLock())
+                .as("Should have a lock token share referencing to immutable ts lock token")
+                .isInstanceOf(LockTokenShare.class)
+                .extracting(t -> ((LockTokenShare) t).sharedLockToken())
+                .isEqualTo(batchedStartTransactionResponse.immutableTimestamp().getLock());
+
+        assertThat(startTransactionResponse.immutableTimestamp().getImmutableTimestamp())
+                .as("Should have same immutable timestamp")
+                .isEqualTo(batchedStartTransactionResponse.immutableTimestamp().getImmutableTimestamp());
+
+        assertThat(startTransactionResponse.startTimestampAndPartition().partition())
+                .as("Should have same partition value")
+                .isEqualTo(batchedStartTransactionResponse.timestamps().partition());
+
+        assertThat(batchedStartTransactionResponse.timestamps().stream())
+                .as("Start timestamp should be contained by batched response")
+                .contains(startTransactionResponse.startTimestampAndPartition().timestamp());
+    }
+
+
+    private static StartTransactionResponseV4 getStartTransactionResponse(
+            long lowestStartTs, int count) {
+        return StartTransactionResponseV4.of(
+                IMMUTABLE_TS_RESPONSE,
+                getPartitionedTimestamps(lowestStartTs, count),
+                LEASE);
+    }
+
+    private static PartitionedTimestamps getPartitionedTimestamps(long startTs, int count) {
+        return ImmutablePartitionedTimestamps.builder()
+                .start(startTs)
+                .count(count)
+                .interval(NUM_PARTITIONS)
+                .build();
+    }
+
+    @Value.Immutable
+    interface TestBatchElement extends BatchElement<Void, StartIdentifiedAtlasDbTransactionResponse> {
+        @Override
+        @Nullable Void argument();
```

Comment:
is this only here so you can specify `@Nullable`?

Reply:
yep :(